qutebrowser/utils/utils.py

```diff
@@ -864,6 +864,63 @@ def unused(_arg):
     pass
 
 
+def parse_numsettxt_into_list(txtset, nummax):
```

Comment:
Easiest way to optimize this is probably just saving the range-set as just that: a set of ranges. Then you can check whether a download is included in `O(r)`, *r* being the number of ranges (thank you Python 3 for constant time range inclusion checks), As each range had to be typed manually, there can't be much. As we check each download, total would be `O(d*r)`, *d* being the number of downloads.

Saving as a set of ranges would be `set(range(1, 2), range(2, 3), range(4,1000000))` for `1,2,4-999999`.